src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export function getFilteredTodo(
+  todos: Todo[],
+  filterTypeTodo: string,
+  query: string,
+) {
+  const filterTodoBy = todos.filter((todo) => {
+    switch (filterTypeTodo) {
+      case 'all':
+        return todo;
+
+      case 'completed':
+        return todo.completed === true;
+
+      case 'active':
+        return todo.completed === false;
+
+      default:
+        return 0;
+    }
+  });
+
+  return filterTodoBy.filter(({ title }) => (
+    title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+  ));
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState(0);
```

Comment:
Default value for selectedTodo is 0 💩
```suggestion
  const [selectedUserId, setSelectedUserId] = useState(null);
  const [selectedTodo, setSelectedTodo] = useState(null);
```